core/domain/takeout_service_test.py

```diff
@@ -608,3 +651,43 @@ def test_export_data_nontrivial(self):
 
         exported_data = takeout_service.export_data_for_user(self.USER_ID_1)
         self.assertEqual(exported_data, expected_export)
+
+
+    def test_get_models_which_should_be_exported(self):
+        """Ensure that the set of models to export is the set of models with
+        export policy CONTAINS_USER_DATA, and that all other models have
+        export policy NOT_APPLICABLE.
+        """
+        all_models = []
+        model_names_list = [
+            model_name
+            for model_name in dir(models.NAMES)
+            if not model_name.startswith('__') and model_name != 'base_model'
+        ]
+        model_modules = models.Registry.import_models(model_names_list)
+        for model_module in model_modules:
+            for _, obj in inspect.getmembers(model_module):
+                if inspect.isclass(obj):
+                    all_models.append(obj)
+
+        models_with_export = (takeout_service
+                              .get_models_which_should_be_exported())
+        unimplemented_models_count = 0
+        for model in all_models:
+            export_policy = model.get_export_policy()
+            if model in models_with_export:
+                self.assertEqual(
+                    base_models.EXPORT_POLICY.CONTAINS_USER_DATA,
+                    export_policy
+                )
+            elif export_policy == base_models.EXPORT_POLICY.TO_BE_IMPLEMENTED:
+                unimplemented_models_count += 1
+            else:
+                self.assertEqual(
+                    base_models.EXPORT_POLICY.NOT_APPLICABLE,
+                    export_policy
+                )
+        # TODO(#8523): This number should be reduced to zero. This number
+        # should not be changed by developers under any circumstance.
+        # Contact @varun-tandon for more information.
+        self.assertEqual(unimplemented_models_count, 24)
```

Comment:
Actually -- instead of checking the length, make a list and ensure that the list is equal to the expected one. Don't want to accept PRs where someone fixes one model in the list and adds another unfixed model to it, thus keeping the count the same.

Reply:
Done.